src/OneExplorer/OneExplorer.ts

```diff
@@ -113,6 +146,179 @@ class Node {
+  readonly type = NodeType.directory;
+
+  readonly openViewType = undefined;
+  icon = vscode.ThemeIcon.Folder;
+  canHide = false;
+
+  constructor(
+      uri: vscode.Uri, icon: vscode.ThemeIcon = vscode.ThemeIcon.Folder, canHide: boolean = false) {
+    super(uri);
+
+    this.icon = icon;
+    this.canHide = canHide;
+  }
+
+  /**
+   * Build a sub-tree under the node
+   *
+   * directory          <- this
+   *   ∟ baseModel      <- children
+   *      ∟ config
+   *         ∟ product
+   */
+  buildChildren = (): void => {
+    const files = fs.readdirSync(this.path);
+
+    for (const fname of files) {
+      const fpath = path.join(this.path, fname);
+      const fstat = fs.statSync(fpath);
+
+      if (fstat.isDirectory()) {
+        const dirNode = NodeFactory.create(NodeType.directory, fpath);
+
+        if (dirNode && dirNode.childNodes.length > 0) {
+          this.childNodes.push(dirNode);
+        }
+      } else if (
+          fstat.isFile() &&
+          (fname.endsWith('.pb') || fname.endsWith('.tflite') || fname.endsWith('.onnx'))) {
+        const baseModelNode = NodeFactory.create(NodeType.baseModel, fpath);
+
+        this.childNodes.push(baseModelNode);
+      }
+    }
+  };
+}
+
+class BaseModelNode extends Node {
+  readonly type = NodeType.baseModel;
+
+  openViewType: string|undefined = undefined;
+  icon = vscode.ThemeIcon.File;
+  canHide = false;
+
+  constructor(
+      uri: vscode.Uri, openViewType: string|undefined = undefined,
+      icon: vscode.ThemeIcon = vscode.ThemeIcon.File, canHide: boolean = false) {
+    super(uri);
+
+    this.openViewType = openViewType;
+    this.icon = icon;
+    this.canHide = canHide;
+  }
+
+  /**
+   * Build a sub-tree under the node
+   *
+   * directory
+   *   ∟ baseModel      <- this
+   *      ∟ config      <- children
+   *         ∟ product
+   */
+  buildChildren = (): void => {
+    const configPaths = getCfgList().filter(cfg => {
+      const cfgObj = ConfigObj.createConfigObj(vscode.Uri.file(cfg));
+      if (!cfgObj) {
+        Logger.info('OneExplorer', `Failed to open file ${cfg}`);
+        return false;
+      }
+
+      return cfgObj.isChildOf(this.path);
+    });
+
+    configPaths.forEach(configPath => {
+      const configNode = NodeFactory.create(NodeType.config, configPath);
+
+      this.childNodes.push(configNode);
+    });
+  };
+}
+
+class ConfigNode extends Node {
+  readonly type = NodeType.config;
+
+  openViewType = 'cfg.editor';
+  icon = new vscode.ThemeIcon('gear');
+  canHide = false;
+
+  constructor(
+      uri: vscode.Uri, openViewType: string = 'cfg.editor',
+      icon: vscode.ThemeIcon = new vscode.ThemeIcon('gear'), canHide: boolean = false) {
+    super(uri);
+
+    this.openViewType = openViewType;
+    this.icon = icon;
+    this.canHide = canHide;
+  }
+
+  /**
+   * Build a sub-tree under the node
+   *
+   * directory
+   *   ∟ baseModel
+   *      ∟ config      <- this
+   *         ∟ product  <- children
+   */
```

Comment:
Not to give a pressure to implement, but I'm wondering that do we have to make sure the product is generated by the parent node config or not. 
If two different configs generate two different product with same name, the product node will be presented in child of both configs, but one of them should be wrong. (wrong? fake? false?)
Was there any discussion about it before?

Reply:
> If two different configs generate two different product with same name, the product node will be presented in child of both configs, but one of them should be wrong. (wrong? fake? false?)

There is no such case.

Each config file will have paths of its produced files.
If some files' path are the same, then it means the files are the same.
So you question is: If two different configs generates product with the same name, will the product node will presented as a child of both nodes?
And the answer is naturally, yes :-D